os/kernel/mqueue/mq_send.c

```diff
@@ -123,7 +123,8 @@
  *
  *   EAGAIN   The queue was empty, and the O_NONBLOCK flag was set for the
  *            message queue description referred to by mqdes.
- *   EINVAL   Either msg or mqdes is NULL or the value of prio is invalid.
+ *   EINVAL   Either msg or mqdes is NULL or the value of prio is invalid,
+ * 	 		  or mqdes is present in task group's mq list.
```

Comment:
I think this should be,
```suggestion
 * 	 		  or mqdes is not present in task group's mq list.
```

But here too, I think EBADF is better instead of EINVAL.